home/templatetags/customtags.py

```diff
@@ -19,3 +21,14 @@ def pagination(pages, page):
 @register.filter
 def stream_name(stream_id):
     return dict(STREAM_CHOICES).get(stream_id, 'Unknown')
+
+@register.filter
+def zulip_url(title, stream):
+    """Return the Zulip url given the title. """
+
+    # Some browsers zealously URI-decode the contents of window.location.hash.
+    # So Zulip hides the URI-encoding by replacing '%' with '.'
+    replace = lambda x: quote(x, safe='').replace('.', '%2E').replace('%', '.')
+    hash_path = 'narrow/stream/%s/topic/%s' % (replace(stream), replace(title))
+
```

Comment:
Might be worth documenting for posterity in the comments in the code (as well as on this PR) that this is the Right Way to do URL construction.